techniques/html/H-creating-dialogs-with-the-html-dialog-element.html

```diff
@@ -0,0 +1,215 @@
+		<section class="meta">
+			<p class="id">ID: </p>
+			<p class="technology">Technology: html</p>
+			<p class="type">Type: Technique</p>
+		</section>
+		<section id="applicability">
+			<h2>When to Use</h2>
+      <p>HTML</p>
+		</section>
+		<section id="description">
+			<h2>Description</h2>
+			<p>Site designers often want to create modal dialogs to get users to focus on a task-related activity outside of the rest of the page's content. Although this can be accomplished using <a href="https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/">ARIA to create a modal dialog</a>, the HTML <code class="language-html">dialog</code> element can also be used. As well as meeting <a href="https://www.w3.org/TR/using-aria/#rule1">the first rule of ARIA</a>, the HTML <code class="language-html">dialog</code> element offers several advantages to its ARIA equivalent, with the browser handling these features:</p>
+      <ul>
+        <li>focus management into the newly-opened modal <code class="language-html">dialog</code>;</li>
+        <li>focus management back onto the triggering element (assuming the trigger is still on the page) when the modal <code class="language-html">dialog</code> is closed;</li>
+        <li>focus is constrained to the contents of the modal <code class="language-html">dialog</code> and the browser's chrome;</li>
+        <li>with no extra code, the underlying page is hidden from assistive technology when the modal <code class="language-html">dialog</code> is open;</li>
+        <li>the use of the <kbd>Escape</kbd> key to close the modal <code class="language-html">dialog</code>.</li>
+      </ul>
+      <p>This technique uses the HTML <code class="language-html">dialog</code> element rather than the ARIA equivalent.</p>
+   </section><section id="examples"><h2>Examples</h2>
+      <section class="example">
+         <h3>A dialog to sign up to a mailing list</h3>
+				 <p>This is an example of using a modal <code class="language-html">dialog</code> element to show a mailing-list sign-up form to a user. The main part of the page contains a <code class="language-html">button</code> element that, when activated, triggers the appearance of the modal <code class="language-html">dialog</code>. The <code class="language-html">button</code> uses the <code class="language-html">aria-controls</code> property to create a programmatic connection between the <code class="language-html">button</code> and the <code class="language-html">dialog</code>; the <code class="language-html">aria-haspopup</code> property, set to <code class="language-html">dialog</code>, to tell assistive technology that the <code class="language-html">button</code> controls a <code class="language-html">dialog</code>, and the <code class="language-html">type</code> attribute to tell the browser that this isn't a <code class="language-html">submit</code> <code class="language-html">button</code>.</p>
```

Comment:
I'm not sure this technique should be mentioning either aria-haspopup or aria-controls.  

see note about limiting unnecessary use of `aria-haspopup` in ARIA 1.2 - https://www.w3.org/TR/wai-aria-1.2/#issue-container-generatedID-57   but without a visual design, i'm assuming that based on the markup, there would be no visual indicator that the button would cause a 'popup'.  I'm indifferent to the attribute staying so long as it's prefaced that it's _optionally_ added.

re: aria-controls, there's really no value in the attribute being here, both for users or for UA, since the browser will handle focus of the dialog element - any benefit that aria-controls could have had is negated by the default focus management.   I think this one should be removed.

Reply:
thanks for that feedback—I hadn't seen the ARIA 1.2 note on `aria-haspopup`. I removed `aria-controls` as well.